contracts/contracts/interfaces/IMessageProcessor.sol

```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+interface IMessageProcessor {
```

Comment:
let's just add Natspec here

Reply:
Do we also need to add Natspec for the functions inside interface contracts? (since I think they would be same as the contract --> e.g. the Natspec of `topup` in **IPoll.sol** would be same as the Natspec in **Poll.sol**)

Reply:
mm what's the standard way of doing things here @crisgarner? 
imo we can just copy paste?

Reply:
I've seen the interface add the Natspec and the implementation uses inheritance notes:

```solidity

interface IPoll {

/// @notice do something
function doSomething() external;

}

contract Poll is IPoll {

/// @inheritdoc IPoll
function doSomething() external { ... }

}

```

Reply:
https://docs.soliditylang.org/en/latest/natspec-format.html#inheritance-notes

Reply:
So do you recommend to write full Natspec in interfaces and inheritdoc in contracts? (or doing oppositely?

Reply:
> I've seen the interface add the Natspec and the implementation uses inheritance notes when we want to inherit natspec from another contract

Seems like a good approach to me 👍